### ISSUE TITLE
Import data from Llave del Saber

### ISSUE THREAD

**Issue by barbayellow** [ideascube/ideascube#662]
The National Colombian Library gave us a sample of the data we must import. I can upload it here because it's anonymized.
 
We must be able to import these data in ideascube and specifically in the biblioteca movil (ola Hombre) project. 

In the sample uploaded here, we need to import:

* Llave
* Sexo
* Nivel Educativo
* Edad
* Discapacidades
* Ocupaciones

Those we don't need:
* Etnia
* Estrato
* Correo Electrónico
* Recibir Correo

There is a mapping on the 2nd tab of the file.
[llavedelsaber.xlsx](https://github.com/ideascube/ideascube/files/658877/llavedelsaber.xlsx)

This should replace #609

**Comment by bochecha** [comment]
Ocupaciones can have multiple values in LDS, we decided to only take the first value.

We will need a mapping for the values, though.

**Comment by barbayellow** [comment]
1. No tiende means "no education, I did not go to school" 
2. Edad : we really need this info. Could we imagine a kind of mathematic conversion? Because tha age is an information that changes every year anyway. and we don't need that.
3. + 4. I've asked for a direct access to llave del saber online. I've also asked for the correct translations.

**Comment by bochecha** [comment]
> Could we imagine a kind of mathematic conversion?

No we can't.

Someone who is 17 today could be born in 2000 or in 1999.

We might be ok with this (small) loss of information though?

**Comment by barbayellow** [comment]
Yes, we are totally fine with that. Data will only be used for impact evaluation. Which mean we will certainly use estimations and ranges (15 to 20) rather than exact age.

**Comment by barbayellow** [comment]
I got the translation for the data > 
- DOCENTE : teacher
- INDEPENDIENTE : self employed
- PENSIONADO : Retired
- DESEMPLEADO unemployed
- ESTUDIANTE : student
- NINGUNA, : none

**Comment by bochecha** [comment]
> DESEMPLEADO unemployed

**Comment by barbayellow** [comment]
"None" means... well, I don't really knox. IUt could be "we have no information" or "I have no situation". I'm asking this to our partners.

**Comment by barbayellow** [comment]
I got access to llabe del saver and here are the values for the fields : 

**Nivel educativo**
![image](https://cloud.githubusercontent.com/assets/4816690/21857825/1bfaaed0-d827-11e6-9deb-5c3dfb952df5.png)

**Occupacion**
![image](https://cloud.githubusercontent.com/assets/4816690/21857893/6208a8b4-d827-11e6-9224-60ebc8a1bd3d.png)

**Sexo**
![image](https://cloud.githubusercontent.com/assets/4816690/21857930/83caf5e2-d827-11e6-8e23-e5f835838cdc.png)

**Comment by barbayellow** [comment]
and yes, llave del saver uses comic sans ms

**Comment by bochecha** [comment]
@barbayellow Thanks for those, it's super useful.

I'm still missing the list of possible occupations, as well as what "none"/"ninguna" means for occupations.

**Comment by barbayellow** [comment]
My bad! Here are the list of occupaciones 
![image](https://cloud.githubusercontent.com/assets/4816690/21892428/2704a51e-d8d7-11e6-8abb-7547ba69c3b2.png)

And Sergio explained me that "none"/"ninguna"  means "no occupation" which is different from "desempleo" (au chômage).

**Comment by bochecha** [comment]
Thanks @barbayellow .

So now, with all this, I have a few questions left.

## Occupations

Currently in Ideascube, we have the following occupations defined:

* Student
* Teacher
* Without profession
* Profit profession
* Volunteering
* Homemaker
* Retired
* Other

This doesn't map well to the Llave del Saber options:

1.  We don't distinguish between "no occupation" and "au chômage".
    Should we? Or do we map both "ninguna" and "desempleo" to our single "Without profession"?
2.  What does "oficios del hogar" mean? What should we map it to?
3.  What does "investigador" mean? What should we map it to?
4.  What is the "Profit profession" we currently have?
5.  We have some options (e.g Volunteering and Homemaker) which don't exist in Llave del Saber. Therefore, they probably use "Otro" for these, which means we'll map them to our "Other" on import.
    Are we ok with this?

## School level

Currently in Ideascube we have the following school levels define:

* Primary
* Secondary
* Professional
* Higher education

For the purpose of this change, I'm adding: "Did not go to school"

This maps quite neatly to the ones in Llave del Saber:

* "No tiene" becomes "Did not go to school"
* "Primaria" becomes "Primary"
* "Secundaria" becomes "Secondary"
* "Universitario" becomes "Higher education"

I only have doubts on the following two:

* Does "Técnico" map to our "Professional"?
* Do we map "Postgrado" to "Higher education"? Or should I add a new "Postgraduate" level?

--- 

The code is pretty much ready, but those are all functional issues I'm having trouble wrangling.

**Comment by barbayellow** [comment]
The main idea is that we must be able to recreate a set of data from ideascube that will match with llabe del saber.  The only information we are ok to loose - since it's technically complicated - is the fact that it's a multiple entries field. In ideascube, we assume the fact that we'll have only one value for this field. 

Sooooo:  

>    We don't distinguish between "no occupation" and "au chômage". Should we? Or do we map both "ninguna" and "desempleo" to our single "Without profession"?
If at some time our Colombian partners decide to switch to ideascube, we must be able to provide an accurate mapping for them. So we should distinguish :  
- ninguna  : none
- desemplea : unemployed

> What does "oficios del hogar" mean? What should we map it to?
We should add a "working from home" case

> What does "investigador" mean? What should we map it to?
"chercheur". We must add it. "Researcher"

>    What is the "Profit profession" we currently have?
I guess the logic behind this is volonteer : bénévole and Profit profession : salarié. It's not relevant for llabe. So it's not relevant here. We must therefore add:
- chercheur / researcher
- enseignant / teacher
- employé / employee
- indépendant / freelance
- travail à la maison / working from home

> We have some options (e.g Volunteering and Homemaker) which don't exist in Llave del Saber. Therefore, they probably use "Otro" for these, which means we'll map them to our "Other" on import. Are we ok with this?
-------
The education issue is little bit trickier. We're asking our partner directly.

**Comment by bochecha** [comment]
> We should add a "working from home" case

Is that really "working from home", for example like I do? (I'm a freelancer, my office is inside my home)

Or is that a "homemaker"? (often called "housewife")

If the former, then it really doesn't make sense in our case, it's only a valuable information in conjunction with e.g "freelance" or "employee".

> employé / employee

Is that not the "Profit profession" you explained just a few lines before? 😕

**Comment by barbayellow** [comment]
> Is that not the "Profit profession" you explained just a few lines before? :confused:
I don't think so since Independiente is also a profit profession, aswell as Investigador, Docente.

> Or is that a "homemaker"? (often called "housewife")
Question asked to Sergio.

**Comment by bochecha** [comment]
> I don't think so since Independiente is also a profit profession, aswell as Investigador, Docente.

I agree.

Which is why I think that our "profit profession" was made to encompass all these, and if we are to split out freelance and researchers, then we should **rename** "Profit profession" to "Employee".

**Comment by bochecha** [comment]
Having discussed this over, we agreed that we probably can't map all of the Llave del Saber values to our Ideascube ones, as some really are specific to Colombia, and would be hard to translate/explain in other contexts. (e.g the "técnico" and "técnologo" school levels, which at the very least don't exist here in France)

What we agreed is that wherever it made sense, we would add the values to our list of possible choices.

The interesting ones seem to be:

*   occupation:
    *   No occupation
    *   Researcher
    *   Freelance
*   school levels:
    *   Did not go to school

We also agreed to rename the "Profit profession" occupation into "Employee", as it was originally envisioned to cover things we would now separate (e.g "Researcher" and "Freelance") as well as renaming "Without profession" into "Unemployed", to better distinguish it from the "No occupation" we will add.

Finally, the plan is to remove the `choices` constraint on the `current_occupation` and `school_level` fields, so that any value is accepted.

However, we still want the form widget to provide a default list of values, equivalent to what we currently use as `choices=` constraint.

And once a custom value has been entered for a field, it should get remembered and added to the list of displayed values for the next form submission. This should help guide the staff towards reusing the same values, making the stats more useful.

I've spent most of today trying to make a widget for this.

I originally wanted something like the [GtkComboBoxEntry](http://pygtk.org/pygtk2tutorial/figures/comboboxentrybasic.png), but unfortunately that doesn't exist in HTML. 😞 

So I tried playing with [the `<datalist>` thing](http://stackoverflow.com/a/22805615/6528176), but that's not supported on Safari (OS X and iOS), and it just doesn't work well anyway for us. (because our options have both a value and a label)

All in all, the best I could do was this:

![screenshot from 2017-01-13 19-34-05](https://cloud.githubusercontent.com/assets/272170/21941063/4abd04f6-d9c7-11e6-9b12-3f04be4ca434.png)

I'm not sure staff would find it clear how they are supposed to interact with it though...

@barbayellow What do you think?

**Comment by barbayellow** [comment]
I think this is what we need ! Maybe we should start thinking about adding some explanations text alongside this kind of feature. But this is something we'll have to talk about in the scope of a specific work on UI. 
So let's go!

**Comment by bochecha** [comment]
On the train yesterday I ended up rethinking my approach to this widget, and I got this:

![screenshot from 2017-01-15 11-24-01](https://cloud.githubusercontent.com/assets/272170/21961848/368a368a-db15-11e6-9c0f-ca2dc4f3d908.png)

It looks like a combobox with a text entry. Click that button and behold:

![screenshot from 2017-01-15 11-24-16](https://cloud.githubusercontent.com/assets/272170/21961850/39dc28ca-db15-11e6-89d1-1c1b9e296148.png)

It requires Javascript though. Is that ok for the user new/edit page?
